### PR TITLE
WIP: Federation move out of core

### PR DESCRIPTION
**Pull request by irfanurrehman**
First PR on kubernetes/federation :)
cc @marun @shashidharatd

### CODE REVIEW

File: hack/sync-from-kubernetes.sh [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# This script is to be used in the interim to pull any new commits relevant to federation in k8s.io/kubernetes.
+# It takes direct inspiration from https://github.com/kubernetes/apimachinery/blob/master/hack/sync-from-kubernetes.sh
+
+set -e
```
Comment (shashidharatd) [inline hack/sync-from-kubernetes.sh]: Instead could use the following
```
set -o errexit
set -o nounset
set -o pipefail
```
This is pretty much the standard followed in kubernetes for all shell scripts.

File: hack/last-fed-sha [hunk 2]
```diff
@@ -0,0 +1 @@
+40ab598d06835906b0e039d1f92a5f10ef079759
```
Comment (shashidharatd) [inline hack/last-fed-sha]: better to keep these files along with the script in some specific directory within hack directory. In future we may need to store additional files/scripts under hack, which may get mixed up.

**Comment by irfanurrehman** [comment]
thanks @shashidharatd for having a look.
Updated!

**Comment by sttts** [comment]
Doesn't this destroy the federation commit history? Please use `git filter-branch` and a direct push to preserve it.

**Comment by sttts** [comment]
Or maybe it's just github that displays it this chaotically.

**Comment by irfanurrehman** [comment]
@sttts, thanks for having a look. Actually we have pushed a set of commits to branch `fed-move-out` using filter-branch. We were a little apprehensive pushing all those commits to master, until we are ready will all the pieces for a workable code here. This PR just reflects the current state of our work (for collaborating with others) and will probably not be merged in its current shape.

**Comment by irfanurrehman** [comment]
closing this to raise the same against master!

**Comment by irfanurrehman** [comment]
Closing this to raise a clean PR with only needed changes.

**Pull request closed without merging**